torch/csrc/jit/tensorexpr/eval.cpp

```diff
@@ -672,13 +672,47 @@ class SimpleIREvaluatorImpl : public IRVisitor {
     return {};
   }
 
+  template <typename T>
+  void check_bounds(T v) {
```

Comment:
The reason I used templating was for error handling, i.e. 
```
      throw malformed_input(
          "dimensions and indices mismatch in check_bounds", v);
```
and
```
          throw malformed_input(ss.str(), v);
```

Is there another good way to have helpful errors here?

Reply:
If I understand correctly, we can throw an exception in any code areas without using templating. 
Here's an example (exception thrown in a function):
https://github.com/pytorch/pytorch/blob/3b27304d20f53619344b74e683357424ca3d4029/torch/csrc/jit/tensorexpr/eval.cpp#L407

Reply:
I think printing `v->buf` would provide enough information for the error message to be helpful, but if you'd like to print Store/Load containing it, we could change `check_bounds` to return bool instead of raising an exception and then raise at the call site.